opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableEdition.tsx

```diff
@@ -0,0 +1,159 @@
+import { stixCyberObservableEditionOverviewFocus } from './StixCyberObservableEditionOverview';
+import useHelper from '../../../../utils/hooks/useHelper';
+import { useFormatter } from '../../../../components/i18n';
+import ThemeDark from '../../../../components/ThemeDark';
+import ThemeLight from '../../../../components/ThemeLight';
+import { QueryRenderer } from '../../../../relay/environment';
+import StixCyberObservableEditionContainer from './StixCyberObservableEditionContainer';
+import Loader, { LoaderVariant } from '../../../../components/Loader';
+import EditEntityControlledDial from '../../../../components/EditEntityControlledDial';
+import { StixCyberObservableEditionContainerQuery$data } from './__generated__/StixCyberObservableEditionContainerQuery.graphql';
+
+export const stixCyberObservableEditionQuery = graphql`
+  query StixCyberObservableEditionContainerQuery($id: String!) {
+    stixCyberObservable(id: $id) {
+      ...StixCyberObservableEditionContainer_stixCyberObservable
+      ...StixCyberObservable_stixCyberObservable
+    }
+  }
+`;
+
+interface StixCyberObservableEditionProps {
+  stixCyberObservableId: string,
+  open?: boolean,
+  handleClose?: () => void,
+}
+
+const StixCyberObservableEdition: FunctionComponent<StixCyberObservableEditionProps> = ({
+  stixCyberObservableId,
+  open: graphOpen,
+  handleClose: handleGraphClose,
+}) => {
+  const { t_i18n } = useFormatter();
+  const [open, setOpen] = useState<boolean>(false);
+  const [commit] = useApiMutation(stixCyberObservableEditionOverviewFocus);
+  const { isFeatureEnable } = useHelper();
+  const isFABReplaced = isFeatureEnable('FAB_REPLACEMENT');
+  const { palette: { mode } } = useTheme();
+  const theme = mode === 'dark'
+    ? ThemeDark()
+    : ThemeLight();
+
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => {
+    commit({
+      variables: {
+        id: stixCyberObservableId,
+        input: { focusOn: '' },
+      },
+    });
+    setOpen(false);
+  };
+  const transition = theme.transitions?.create
+    ? theme.transitions.create('width', {
+      easing: theme.transitions.easing?.sharp,
+      duration: theme.transitions.duration?.enteringScreen,
+    })
+    : undefined;
+
+  const renderClassic = () => (
+    <>
+      {isFABReplaced
+        ? (
+          <EditEntityControlledDial onOpen={handleOpen} onClose={() => {}}/>
+        ) : (
+          <Fab
+            onClick={handleOpen}
+            color="primary"
+            aria-label={t_i18n('Update')}
+            style={{
+              position: 'fixed',
+              bottom: 30,
+              right: 30,
+            }}
+          >
+            <Edit />
+          </Fab>
+        )
+      }
+      <Drawer
+        open={open}
+        anchor="right"
+        sx={{
+          zIndex: 1202,
+          minHeight: '100vh',
+          width: '50%',
+          position: 'fixed',
+          overflow: 'auto',
+          transition,
+          padding: 0,
+        }}
```

Comment:
the style is not good on the drawer, for instance it does not take 50% of the screen width.

It's because the style should actually be applied to the Paper inside the Drawer (see the original JS code, the class was applied onto `paper` field in `classes`


```suggestion
        paperProps={{
          sx: {
            zIndex: 1202,
            minHeight: '100vh',
            width: '50%',
            position: 'fixed',
            overflow: 'auto',
            transition,
            padding: 0,
          }
        }}
```